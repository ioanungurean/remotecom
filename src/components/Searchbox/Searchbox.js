import { StyledSearchbox, StyledInput, StyledIconSearch } from './Searchbox.styled';

const Searchbox = ({ value, placeholder, onChange }) => {
  const handleChange = (ev) => {
    onChange(ev.target.value);
  };

  return (
    <StyledSearchbox htmlFor="search">
      <StyledIconSearch title="Search employees" />
      <StyledInput id="search" value={value} placeholder={placeholder} onChange={handleChange} />
    </StyledSearchbox>
  );
};

export default Searchbox;
