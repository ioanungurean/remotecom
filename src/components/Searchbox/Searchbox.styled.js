import styled from 'styled-components';
import { ReactComponent as IconSearch } from 'theme/icons/search.svg';

export const StyledInput = styled.input`
  font-size: 14px;

  padding: 0;

  color: var(--colors-darkBlue);
  border: none;
  outline: none;
  background: none;
  box-shadow: none;
  ::placeholder {
    color: var(--colors-lynch);
  }
`;

export const StyledSearchbox = styled.label`
  display: flex;

  padding: 0 16px;

  border: 1px solid transparent;
  border-radius: 20px;

  &:hover,
  &:focus-within {
    border: 1px solid var(--colors-selago);
  }

  &:hover {
    background-color: var(--colors-zircon);
  }

  &:focus-within {
    box-shadow: inset 1px 2px 3px var(--colors-coldIceGrey);
  }
`;

export const StyledIconSearch = styled(IconSearch)`
  margin-right: 12px;
  fill: var(--colors-lynch);
`;
