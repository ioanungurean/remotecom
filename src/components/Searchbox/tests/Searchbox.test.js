import { render, screen, fireEvent } from '@testing-library/react';
import Searchbox from '../index';

describe('Searchbox', () => {
  it('renders correctly', () => {
    const { container } = render(<Searchbox placeholder="Test" />);

    expect(container).toMatchSnapshot();
  });

  it('shows the correct value', () => {
    const value = 'Searchbox';
    render(<Searchbox placeholder="Searchbox" value={value} />);

    expect(screen.getByRole('textbox')).toHaveValue(value);
  });

  it('calls the onChange handler', () => {
    const value = 'test';
    const spy = jest.fn();
    render(<Searchbox placeholder="Searchbox" onChange={spy} />);

    const input = screen.getByRole('textbox');
    fireEvent.change(input, { target: { value } });

    expect(input.value).toBe(value);
  });
});
