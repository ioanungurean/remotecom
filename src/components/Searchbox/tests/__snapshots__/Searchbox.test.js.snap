// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbox renders correctly 1`] = `
<div>
  <label
    class="sc-gsTCUz cPmsHZ"
    for="search"
  >
    <svg
      class="sc-dlfnbm gWdAtF"
      title="Search employees"
    >
      search.svg
    </svg>
    <input
      class="sc-bdfBwQ jETauC"
      id="search"
      placeholder="Test"
      value=""
    />
  </label>
</div>
`;
